packages/auth/lib/index.d.ts

```diff
@@ -472,7 +472,21 @@ export namespace FirebaseAuthTypes {
   /**
    * Contains information about a second factor.
    */
-  export interface MultiFactorInfo {
+  export type MultiFactorInfo = PhoneMultiFactorInfo | TotpMultiFactorInfo;
+
+  export interface PhoneMultiFactorInfo extends MultiFactorInfoCommon {
+    factorId: 'phone';
+    /**
+     * The phone number used for this factor.
+     */
+    phoneNumber: string;
```

Comment:
You never know, and per the native idiom in iOS anyway, it *might* return null

```suggestion
    phoneNumber: string | null;
```

Reply:
I'm somewhat reluctant to mark this as nullable based on https://github.com/invertase/react-native-firebase/pull/7565#discussion_r1465428041 - whilst anything is possible in javascript, I believe the intention of the SDK is that this is non-nullable.

If you feel it's essential to mark as nullable then I can still

Reply:
good defense - resolved via a quick look at firebase-js-sdk types to make 100% certain, https://github.com/search?q=repo%3Afirebase%2Ffirebase-js-sdk%20phonemultifactorinfo&type=code 

it is not nullable ✅